include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -227,4 +231,151 @@ std::string get_cigar_string(alignment_type && alignment,
                             query_start_pos, query_end_pos, extended_cigar);
 }
 
+/*!\brief Transforms a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+ * \ingroup alignment_file
+ *
+ * \tparam soft_clipping_begin_type  The type of the soft_clipping_begin value; Must model std::Integral.
+ * \tparam soft_clipping_end_type    The type of the soft_clipping_end value; Must model std::Integral.
+ * \tparam cigar_view_type           The type of a single pass input view over the cigar string.
+ *
+ * \param[out] soft_clipping_begin  Set to the soft clipping at the beginning of the alignment.
```

Comment:
oh yes this was misleading. It will be set to the NUMBER of soft clipped bases at the begin and end of the alignment. I changed the documentation and variable names accordingly.